pkg/agent/packetcapture/bpf/parser.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bpf
+
+import (
+	"encoding/binary"
+	"net"
+	"strings"
+
+	"golang.org/x/net/bpf"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+)
+
+// CompilePacketFilter compile the CRD spec to bpf instructions. For now, we only focus on
+// ipv4 traffic. Compare to the raw BPF filter supported by libpcap, we only need to support
+// limited user cases, so an expression parser is not needed.
+func CompilePacketFilter(packetSpec *crdv1alpha1.Packet, srcIP, dstIP net.IP) []bpf.Instruction {
+	size := uint8(calInstructionsSize(packetSpec))
+
+	// ipv4 check
+	inst := []bpf.Instruction{loadEtherKind}
+	// skip means how many instructions we need to skip if the compare fails.
+	// for example, for now we have 2 instructions, and the total size is 17, if ipv4
+	// check failed, we need to jump to the end (ret #0), skip 17-3=14 instructions.
+	// if check succeed, skipTrue means we jump to the next instruction.
+	inst = append(inst, compareProtocolIP4(0, size-3))
```

Comment:
I don't quite get why it's `size-3` when the check failed. I suppose we should count the remained instructions' size and jump to the end? Maybe explain where the 3 comes from.

Reply:
Ok. `3` means there are 3 instructions by far. will update this.